include/fc/container/flat.hpp

```diff
@@ -86,11 +86,16 @@ namespace fc {
           --_max_depth;
           unsigned_int size;
           unpack( s, size, _max_depth );
-          value.resize( size );
           if( !std::is_fundamental<T>::value ) {
-             for( auto& item : value )
-                unpack( s, item, _max_depth );
+             value.resize( std::min( size.value, FC_MAX_PREALLOC_SIZE ) );
+             for( uint64_t i = 0; i < size; i++ )
+             {
+                if( i >= value.size() )
+                   value.resize( std::min( 2*value.size(), size.value ) );
```

Comment:
Perhaps change `2*value.size()` to `value.size() + FC_MAX_PREALLOC_SIZE` to be more defensive? Which is better?

Reply:
(Presumably) `resize` can cost O(n), so with linear growth you arrive at O(n^2) for large n, whereas exponential growth will result in O(n*log(n)).

Reply:
Also note that the vector will only be resized to 2*n if the attacker has already sent n items.